### ISSUE TITLE
February Iteration Plan

### ISSUE THREAD

**Issue by egamma** [Microsoft/vscode#2616]
February is our last full iteration before we start the end game for the [Build 2016](http://build.microsoft.com/) milestone at the end of March.

We made good progress in the areas such as accessibility and localization, but more work needs to happen. Therefore we arrived at the following themes for February:
* Continue the work on accessibility, localization.
* Reduce adoption blockers - the item we want to tackle is support for *Code Folding*, the most voted for [feature request](https://visualstudio.uservoice.com/forums/293070-visual-studio-code/suggestions/7752321-add-code-folding-support), Issue #105.
* Reduce the engineering debt.
* Respond to feedback and try to address as many issues as possible.

For more information about our development process please refer to the development process [page](https://github.com/Microsoft/vscode/wiki/Development-Process).

##### Monday, Feb 22
- [x] Code freeze at end of day

##### Tuesday
- [x] Test build done
- [x] Testing

##### Wednesday
- [x] Testing
- [x] Fixing
- [x] Omnisharp Smoke Test @weinand 
- [x] Run OSS tool @isidorn 

##### Thursday
- [x] Fixing
- [x] Verification

##### Friday
- [x] Fixing
- [x] Smoketest
  - [x] Windows @alexandrudima 
  - [x] OS X @isidorn
  - [x] Linux @aeschli
- [x] All release notes updated @weinand 
- [ ] All issues verified
- ~~Signing process initiated @joaomoreno~~ Not needed, we can now sign ourselves.
- [ ] Satellite modules ready, version updated, smoke tested
  - [ ] vscode @bpasero 
  - [ ] yo generator @aeschli 
  - [x] vsce @joaomoreno: published vsce@1.2.0
  - [ ] Debug adapter @weinand 
  - [ ] Language client/server @dbaeumer 
  - [ ] eslint (recommended linter for JS) @dbaeumer 

##### Monday, Feb 29
- [ ] Release [`insider`](https://blogs.msdn.microsoft.com/vscode/2016/02/01/introducing-insiders-builds/) channel

##### Monday, March 7
- [ ] Release `stable` channel

### Plan Items

| Mark  | Description |
| ------------- | ------------- |
| :runner:  | work in progress  |
| :hand:  | task is blocked  |
| :muscle:  | stretch goal for this iteration  |

#### Feedback

Issues to address feedback from the community:
- [x] Support folding in the editor #105 @aeschli 
  - Test: #2994 
  - [x]  Release Notes
- [x] Editor - Supporting ruler settings #1673 @alexandrudima 
  - Test: #3233
  - [x]  Release Notes
- [x] Editor - Improve tab and space insertion handling, show what tab size is guessed, support to convert tabs to space and vice versa #1228 @isidorn 
  - Test: #3234
  - [x] Release Notes
- [x] Editor - Add EOL option #2957 @alexandrudima 
  - [x] Release Notes
- [x] UI - Provide action to "Open definition to the side" #2660 (see also #2646) @bpasero
  - Test: #3236
  - [x]  Release Notes
- [x] UI - "Close all but this" context menu option for working files #2643 @bpasero
  - Test: #3235 
  - [x]  Release Notes
- [x] UI - Support to switch to a working file from the quick box #441 @bpasero
  - Test: #2791
  - [x]  Release Notes
- [x] UI - Experimental support for recommending extensions based on usage/setup @jrieken 
  - Test: #2734
  - [x]  Release Notes
- [x] UI - Improve the suggest widget performance #1923 @joaomoreno 
  - Test: #3237
  - [x]  Release Notes
- [x] Debug - Failed preLaunchTask's should abort debugger launch #2370 @isidorn 
  - Test: #2940 
  - [x]  Release Notes
- [x] Debug - Support asp.net debug scenario (was #2263)
  - ~~Add snippet support to tasks.json #3122 @dbaeumer~~
    - Test: #3159
  - [x] Offer the user to configure a task when the 'preLaunchTask' isn't found #3081 @isidorn
    - Test: #2940
    - [x] Release Notes
- [x] Debug/Node - Support `nodemon` development setup by adding support to automatically re-attach when node is restarted #2103 @weinand 
  - Test: #3238
  - [x]  Release Notes
- [x] Debug - Support 'EvaluateRequest' based data hovers #2012 @isidorn  
  - Test: #3245
  - [x]  Release Notes
- [x] Debug - API for extensions needs to support selection of 32-bit or 64-bit debug host #1696 @isidorn 
  - [x]  Release Notes
- *shifted* Tasks - Cannot define multiple commands in tasks.json #981 @dbaeumer 
- [x] Install - Improve install experience on Linux #2679, #269  @Tyriar 
  - Test: #3239
  - [x]  Release Notes
- [x] Launch - Improve launching on Unix based systems #378, #77 @Tyriar 
  - Test: #3240
  - [x]  Release Notes
- [x] Launch - Improving launching of OS X remove need to edit the .bashrc files @joaomoreno
  - Test: #3241
  - [x]  Release Notesd 

#### Debt Items
This section lists items to reduce our engineering debt:
- [x] TSLint adoption
  - [x] clean lint warnings #2102 @all
  - [x] run tslint checks during pre-commit @egamma 

#### Adoptions
We continuously have to adopt components which we require:
- :hand: Adopt the latest electron 0.36.7  @bpasero 
  - Blocked due to atom/electron#4299
- TS 
  - [x] 1.8 @dbaeumer
    - ~~Test~~
  - [x] 1.8.2 (will be available Feb 22nd) @joaomoreno
   - [x] Release Notes  (we should mention that we now include 1.8.2)

#### Engineering Items
This section lists several engineering/development items that the team identified:
- [x] JavaScript - Salsa migration
 - [x] Make Salsa the default JavaScript support and guide users to the release notes #3080 @dbaeumer, @egamma
  - Test: #3170 
  - [x] Release Notes @egamma
- [x] C# unbundling
 - [x] guide user to install the separate extension @jrieken 
 - [x]  unbundle OmniSharp #3167 @joaomoreno 
  - Test: #2701 
  - [x]  Release Notes @chrisdias
- [x] Install update - Improve extension selection performance #2621 @joaomoreno
   - Test: #3242
   - [x]  Release Notes   
- ~~Install update - No progress/status while downloading an extension #2835 @joaomoreno~~
   - ~~Test: #3243~~
   -  ~~Release Notes~~
- [x] Install update - Always install a compatible version of the extension if possible #2894 @joaomoreno
   - Test: #3244
   - [x]  Release Notes   
- [x] Telemetry - provide option to opt out of telemetry @SofianHn
  - Test: #3182 
  - [x] Release Notes @joaomoreno
- ~~Telemetry - Gather metrics on Linux distribution #2690 @SofianHn~~
- [x] Telemetry - create a common module for 'first party' extensions @SofianHn 
  - Test: #3181
  - ~~Release Notes? @SofianHn~~
    - @joaomoreno: I have no idea what this is, neither what I should put in the release notes. The test plan item seems irrelevant to the task description.
- [x] Debug - expose the Trace feature for node debug #2609 @weinand
  - ~~Test~~
- [x] Debug - support function breakpoints in protocol Microsoft/vscode-node-debug#35 @weinand 
  - Test: #2832
  - [x] Release Notes @weinand
- [x] Debug - Remove mono-debug from VS Code and move it to market place, and assist users in installing it from there #2668 @weinand
   - Test: #2831 
   - [x]  Release Notes   
- [x] Keyboard accessibility  #2108 @bpasero 
   - Test: #3246
   - [x]  Release Notes   
- [x] Accessible Debugging #2671 @isidorn 
   - Test: #2941
   - [x]  Release Notes   
- [x] Screen reading #3058 @alexandrudima 
   - Test: #3247
   - [x]  Release Notes   
- [x] Localization  #2041 @dbaeumer, @weinand 
   - Test: #3156
   - [x]  Release Notes   
- [x] :muscle: High Contrast support #2711 #3184 @bgashler1 
   - Test: #3248
   - [x]  Release Notes @joaomoreno
- [x] Doc: add new contents based on community blogs @gregvanl 
   - [x] Release Notes @gregvanl   

### Testers

- @joaomoreno
- @isidorn 
- @weinand 
- @alexandrudima 
- @aeschli 
- @seanmcbreen 
- @chrisdias (from Wednesday only)
- @stevencl 
- @Tyriar 
- @bgashler1 (a little bit)
- @gregvanl

**Comment by profiler-bg** [comment]
I'm confused by all the January mentions. Feb schedule should be end of Feb, should it not? Is it due to using the Jan file as template?

**Comment by chrisdias** [comment]
@profiler-bg yes, this is a work in progress :)

**Comment by bpasero** [comment]
@egamma please update the electron one to blocked, due to https://github.com/atom/electron/issues/4299

**Comment by piyushchauhan2011** [comment]
:+1:

**Comment by nonbackwards** [comment]
:+1: thanks

**Comment by zethon** [comment]
#VisualCodeFolding

**Comment by brennanMKE** [comment]
I cannot wait for code folding. I have been working heavily in VS Code over the past month and with large source files it is so helpful to collapse code to make it more readable.

**Comment by StfBauer** [comment]
:+1:

**Comment by nyn3x** [comment]
:+1:

**Comment by BazIdle** [comment]
Not sure if I should address this here, but:
Any plans on adopting new stuff from [OmniSharp](https://github.com/OmniSharp/omnisharp-roslyn/) ?

[OmniSharp crashes](https://github.com/Microsoft/vscode/issues/1535) are still present for me in the latest (recent, "January"-one) build.

As far as I understand (from OmniSharp issuetracker), they had a lot of major changes recently.

Would love to know an insight ;)

**Comment by leeroya** [comment]
Brilliant Thanks. cant wait.

**Comment by egamma** [comment]
@BazIdle OmniSharp will continue to move forward and it is now in the best hands (the team that also works on Roslyn). The work on Omnisharp for vscode is now tracked in https://github.com/OmniSharp/omnisharp-vscode. I'll move the bug you mention over to this repository.

FYI @DustinCampbell

**Comment by BazIdle** [comment]
@egamma Thanks for the info

**Comment by Figbash** [comment]
@egamma @BazIdle Sorry for bugging you here, but I and many people I know have been struggling with this issue since .10.3, is there any chance someone would be willing to look at it for this iteration?  I have multiple projects that do it and would love to help someone track it down.  There doesn't seem to be any interest in omnisharp-vscode, our issues have been there since December with no chatter.  VSCode cannot be used for us in this state, and I really want to be able to upgrade!
https://github.com/OmniSharp/omnisharp-vscode/issues/6 this is my bug for the same issue.

**Comment by DustinCampbell** [comment]
Sorry @Figbash for not commenting over at https://github.com/OmniSharp/omnisharp-vscode/issues/6. We're in the middle of a bunch of work to move the extension to a new version of OmniSharp that's also in progress. Once we get over that hurdle, I'll dig deeper into your issue.

**Comment by Figbash** [comment]
@DustinCampbell Thank you for letting me know.  No worries about commenting I just wanted to make sure it wasn't lost in the shuffle.

**Comment by Nimzozo** [comment]
:coffee:

**Comment by penguin020** [comment]
This is just great that you make this so transparent.  Thanks.
